.clang-format

```diff
@@ -21,31 +24,82 @@ AllowShortLambdasOnASingleLine: All
 AllowShortIfStatementsOnASingleLine: Never
 AllowShortLoopsOnASingleLine: false
 AlwaysBreakAfterReturnType: TopLevel
-AlwaysBreakBeforeMultilineStrings: false      # (false) true
+AlwaysBreakBeforeMultilineStrings: false
 AlwaysBreakTemplateDeclarations: MultiLine
+
+# Only works wtih macros that do not take parameters
 AttributeMacros:
-BinPackArguments: true                        # (true) false
-BinPackParameters: false                      # (true) false
+ - HSE_ALWAYS_INLINE
+ - HSE_COLD
+ - HSE_CONST
+ - HSE_HOT
+ - HSE_MAYBE_UNUSED
+ - HSE_NOINLINE
+ - HSE_NORETURN
+ - HSE_PACKED
+ - HSE_READ_MOSTLY
+ - HSE_RETURNS_NONNULL
+ - HSE_SENTINEL
+ - HSE_USED
+ - HSE_WARN_UNUSED_RESULT
+ - HSE_WEAK
+
+# TODO:
+# - HSE_LIKELY(_expr)
+# - HSE_UNLIKELY(_expr)
+
+# Treating function attribute macros that take parameters as type macros
+# helps format function delcs that use them.
+TypenameMacros:
+ - HSE_NONNULL
+ - HSE_PRINTF
+ - HSE_ALIGNED
+StatementAttributeLikeMacros:
+StatementMacros:
+ForEachMacros:
+ - cds_list_for_each
+ - cds_list_for_each_entry_rcu
+ - cds_list_for_each_entry_reverse
+ - cds_list_for_each_entry_safe
+ - cds_list_for_each_entry_safe_from
+ - cds_list_for_each_prev
+ - cds_list_for_each_prev_safe
+ - cds_list_for_each_safe
+ - cn_tree_foreach_leaf
+ - cn_tree_foreach_leaf_safe
+ - cn_tree_foreach_node
+ - list_for_each
+ - list_for_each_entry
+ - list_for_each_entry_reverse
+ - list_for_each_entry_reverse_safe
+ - list_for_each_entry_safe
+ - s_list_for_each
+ - s_list_for_each_entry
+ - s_list_for_each_entry_safe
+ - s_list_for_each_safe
+WhitespaceSensitiveMacros:
+BinPackArguments: true
+BinPackParameters: false
```

Comment:
https://stackoverflow.com/questions/29483626/clang-format-force-arguments-and-parameters-to-one-line-each-if-the-statement-o

Reply:
Dunno if that helps, figured you could try it :-)